File: backend/app/schemas/resume.py
from pydantic import BaseModel
from typing import List, Optional


class ResumeResponse(BaseModel):
    id: int
    filename: str
    file_path: str
    user_id: int

    class Config:
        from_attributes = True


class ResumeAnalysis(BaseModel):
    resume_score: int
    ats_score: int

    section_scores: dict
    section_feedback: dict
    ai_rewrite: dict

    skills: List[str]
    education: List[str]
    experience: List[str]
    missing_skills: List[str]

    email: Optional[str] = None
    phone: Optional[str] = None

    suggestions: List[str]

    strengths: List[str]
    weaknesses: List[str]
    improvement_suggestions: List[str]

    # NEW AI Feedback
    ai_feedback: str


class ResumeAnalysisResponse(BaseModel):
    resume: ResumeResponse
    extracted_text: str
    analysis: ResumeAnalysis

File: frontend/src/types/resume.ts
export interface Resume {
  id: number;
  filename: string;
  file_path: string;
  user_id: number;
}

export interface ResumeAnalysis {
  resume_score: number;
  ats_score: number;

  section_scores: Record<string, number>;
  section_feedback: Record<string, string>;
  ai_rewrite: Record<string, string>;

  skills: string[];
  education: string[];
  experience: string[];
  missing_skills: string[];

  email?: string;
  phone?: string;

  suggestions: string[];

  strengths: string[];
  weaknesses: string[];
  improvement_suggestions: string[];

  ai_feedback: string;
}

export interface ResumeAnalysisResponse {
  resume: Resume;
  extracted_text: string;
  analysis: ResumeAnalysis;
}